### DISCUSSION TITLE
Elaborate on known issue: "Programs can behave like there is no OpenGL support when using Mesa `opengl32.dll` since 21.0.0"

### DISCUSSION THREAD

**Discussion by nealkruis** [pal1000/mesa-dist-win#158] [Q&A]
> This is not a defect but rather a behavior change of Mesa when environment variables are misconfigured. It usually happens when selecting a Mesa driver that doesn't exist in release package used or it fails to initialize due to host system not meeting hardware requirements or lacking dependencies. Reading [differences between MSVC and MinGW packages](#differences-between-msvc-and-mingw-packages) and [Mingw and MSVC Package contents](#mingw-and-msvc-package-contents) should aid in troubleshooting.

Can someone explain in more detail the relevant parts of the "[differences between MSVC and MinGW packages](#differences-between-msvc-and-mingw-packages) and [Mingw and MSVC Package contents](#mingw-and-msvc-package-contents)" sections?

I don't see much describing environment variables that can be misconfigured.

**Comment by pal1000** [comment]
The most common scenario for this behavior occurs when you try to explicitly select a Mesa3D OpenGL layered driver via `set GALLIUM_DRIVER=d3d12` or `set GALLIUM_DRIVER=zink`  and your system doesn't meet the respective requirements. See description for GLonD3D12 and zink in [Mingw and MSVC Package contents](https://github.com/pal1000/mesa-dist-win/blob/master/readme.md#mingw-and-msvc-package-contents).
